.github/instructions/uitests.instructions.md

```diff
@@ -199,6 +198,104 @@ public void SoftInputBehaviorTest()
 #endif
 ```
 
+## Running UI Tests Locally
+
+### Quick Test Execution (for rapid development)
+
+When developing and debugging a specific test:
+
+**Android:**
+1. Deploy the TestCases.HostApp:
+   ```bash
+   # Use local dotnet if available, otherwise use global dotnet
+   ./bin/dotnet/dotnet build src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj -f net10.0-android -t:Run
+   # OR:
+   dotnet build src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj -f net10.0-android -t:Run
+   ```
+
+2. Run your specific test:
+   ```bash
+   dotnet test src/Controls/tests/TestCases.Android.Tests/Controls.TestCases.Android.Tests.csproj --filter "FullyQualifiedName~Issue12345"
+   ```
+
+**iOS (3-step process):**
+
+**Step 1: Find iPhone Xs with highest API level**
+```bash
+# View all iPhone Xs devices with their iOS versions
+xcrun simctl list devices available | awk '/^--.*iOS/ {version=$0} /iPhone Xs/ {print version " -> " $0}'
```

Comment:
It's going to be more reliable to get json output, and then transform it using jq:

```
xcrun simctl list devices available --json | jq '...'
```